src/newgrf.cpp

```diff
@@ -1976,34 +1976,34 @@ static ChangeInfoResult StationChangeInfo(uint first, uint last, int prop, ByteR
 					if (buf.HasData(4) && buf.PeekDWord() == 0) {
 						buf.Skip(4);
 						extern const DrawTileSprites _station_display_datas_rail[8];
-						dts->Clone(&_station_display_datas_rail[t % 8]);
+						const DrawTileSprites &dtss = _station_display_datas_rail[t % 8];
+						dts->ground = dtss.ground;
+						dts->seq = dtss.seq;
 						continue;
 					}
 
 					ReadSpriteLayoutSprite(buf, false, false, false, GSF_STATIONS, &dts->ground);
 					/* On error, bail out immediately. Temporary GRF data was already freed */
 					if (_cur.skip_sprites < 0) return CIR_DISABLED;
 
-					static std::vector<DrawTileSeqStruct> tmp_layout;
-					tmp_layout.clear();
 					for (;;) {
-						/* no relative bounding box support */
-						DrawTileSeqStruct &dtss = tmp_layout.emplace_back();
-						MemSetT(&dtss, 0);
+						uint8_t delta_x = buf.ReadByte();
+						if (delta_x == 0x80) break;
 
-						dtss.delta_x = buf.ReadByte();
-						if (dtss.IsTerminator()) break;
-						dtss.delta_y = buf.ReadByte();
-						dtss.delta_z = buf.ReadByte();
-						dtss.size_x = buf.ReadByte();
-						dtss.size_y = buf.ReadByte();
-						dtss.size_z = buf.ReadByte();
+						/* no relative bounding box support */
+						DrawTileSeqStruct &dtss = dts->seq.emplace_back(DrawTileSeqStruct{
```

Comment:
Pretty odd way to emplace. Could we implement the constructor to do it properly? If not it's probably better to do it the way it was before.

Reply:
Technically we don't need to make a constructor, because `emplace_back` also supports value initialization.

However, I wanted to avoid that as designated initializers would be much clearer to show intent. Given this would move/copy, I'll go back to the original way of doing it.
Though not before there's an answer to the question of  the first limitation of #13364. Basically: use inheritance so we can use `std::vector` or `std::span` for the DrawTileSprites' seq, or make everything a `std::vector` and copy things unnecessarily?